pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/TableConfigsRestletResource.java

```diff
@@ -0,0 +1,358 @@
+      Collections.sort(rawTableNames);
+
+      ArrayNode configsList = JsonUtils.newArrayNode();
+      for (String rawTableName : rawTableNames) {
+        configsList.add(rawTableName);
+      }
+      return configsList.toString();
+    } catch (Exception e) {
+      throw new ControllerApplicationException(LOGGER, e.getMessage(), Response.Status.INTERNAL_SERVER_ERROR, e);
+    }
+  }
+
+  /**
+   * Gets the {@link TableConfigs} for the provided raw tableName, by fetching the offline table config for tableName_OFFLINE,
+   * realtime table config for tableName_REALTIME and schema for tableName
+   */
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/tableConfigs/{tableName}")
+  @Authenticate(AccessType.READ)
+  @ApiOperation(value = "Get the TableConfigs for a given raw tableName", notes = "Get the TableConfigs for a given raw tableName")
+  public String getConfig(
+      @ApiParam(value = "Raw table name", required = true) @PathParam("tableName") String tableName) {
+
+    try {
+      Schema schema = _pinotHelixResourceManager.getSchema(tableName);
+      TableConfig offlineTableConfig = _pinotHelixResourceManager.getOfflineTableConfig(tableName);
+      TableConfig realtimeTableConfig = _pinotHelixResourceManager.getRealtimeTableConfig(tableName);
+      TableConfigs config = new TableConfigs(tableName, schema, offlineTableConfig, realtimeTableConfig);
+      return config.toPrettyJsonString();
+    } catch (Exception e) {
+      throw new ControllerApplicationException(LOGGER, e.getMessage(), Response.Status.INTERNAL_SERVER_ERROR, e);
+    }
+  }
+
+  /**
+   * Creates a {@link TableConfigs} using the <code>tableConfigsStr</code>, by creating the schema,
+   * followed by the realtime tableConfig and offline tableConfig as applicable, from the {@link TableConfigs}.
+   * Validates the configs before applying.
+   */
+  @POST
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/tableConfigs")
+  @ApiOperation(value = "Add the TableConfigs using the tableConfigsStr json", notes = "Add the TableConfigs using the tableConfigsStr json")
+  public SuccessResponse addConfig(String tableConfigsStr, @Context HttpHeaders httpHeaders, @Context Request request) {
+    TableConfigs tableConfigs;
+    try {
+      tableConfigs = JsonUtils.stringToObject(tableConfigsStr, TableConfigs.class);
+      validateConfig(tableConfigs);
+    } catch (Exception e) {
+      throw new ControllerApplicationException(LOGGER, String.format("Invalid TableConfigs. %s", e.getMessage()),
+          Response.Status.BAD_REQUEST, e);
+    }
+
+    TableConfig offlineTableConfig = tableConfigs.getOffline();
+    TableConfig realtimeTableConfig = tableConfigs.getRealtime();
+    Schema schema = tableConfigs.getSchema();
+
+    try {
+      String endpointUrl = request.getRequestURL().toString();
+      validatePermissions(schema.getSchemaName(), AccessType.CREATE, httpHeaders, endpointUrl);
```

Comment:
(nit) No need to wrap this one line call into a function, `AccessControl` can be reused here
```suggestion
      AccessControl accessControl = _accessControlFactory.create();
      _accessControlUtils
        .validatePermission(schema.getSchemaName(), AccessType.CREATE, httpHeaders, endpointUrl, accessControl);
      ...
```